src/DGtal/topology/tables/LookUpTableFunctions.h

```diff
@@ -0,0 +1,188 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file LookUpTableFunctions.h
+ *
+ * @author Pablo Hernandez-Cerdan. Institute of Fundamental Sciences.
+ * Massey University. Palmerston North, New Zealand
+ *
+ * @date 2016/03/25
+ *
+ * Generation of Look Up Tables for predicates function in a voxel
+ * neighborhood.
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(LookUpTableFunctions_RECURSES)
+#error Recursive header files inclusion detected in LookUpTableFunctions.h
+#else // defined(LookUpTableFunctions_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define LookUpTableFunctions_RECURSES
+
+#if !defined LookUpTableFunctions_h
+/** Prevents repeated inclusion of headers. */
+#define LookUpTableFunctions_h
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <bitset>
+#include <memory>
+#include "DGtal/topology/Object.h"
+#include "DGtal/helpers/StdDefs.h"
+#include <unordered_map>
+#include "boost/dynamic_bitset.hpp"
+
+namespace DGtal
+{
+  namespace lut {
```

Comment:
Why not a name related to the objective of lut... here topology, homotopy, simplicity, etc ... as you wish 

----- Mail original -----

De: "David Coeurjolly" <notifications@github.com> 
À: "DGtal-team/DGtal" <DGtal@noreply.github.com> 
Cc: "Jacques-Olivier Lachaud" <jacques-olivier.lachaud@univ-savoie.fr> 
Envoyé: Lundi 11 Avril 2016 11:16:01 
Objet: Re: [DGtal-team/DGtal] Add LookUpTableFunctions (#1155) 



In src/DGtal/topology/tables/LookUpTableFunctions.h : 
> +#if !defined LookUpTableFunctions_h
> +/** Prevents repeated inclusion of headers. */
> +#define LookUpTableFunctions_h
> +//////////////////////////////////////////////////////////////////////////////
> +// Inclusions
> +#include <iostream>
> +#include <bitset>
> +#include <memory>
> +#include "DGtal/topology/Object.h"
> +#include "DGtal/helpers/StdDefs.h"
> +#include <unordered_map>
> +#include "boost/dynamic_bitset.hpp"
> +
> +namespace DGtal
> +{
> +  namespace lut { 


no idea, too 

NeigborhoodConfigurations is ok for the filename but I don't know for the namespace... neighborhood:: ? 

— 
You are receiving this because you were mentioned. 
Reply to this email directly or view it on GitHub